test/spinxsk/spinxsk.c

```diff
@@ -1992,6 +2022,182 @@ ProcessPkts(
     return TRUE;
 }
 
+#define MAX_HEADER_STORAGE TCP_HEADER_STORAGE
+
+typedef union _INET_ADDR_STORAGE {
+    IN_ADDR Ipv4;
+    IN6_ADDR Ipv6;
+} INET_ADDR_STORAGE;
+
+//
+// Build and inject frames in the receive path through the FNMP driver.
+//
+UINT8
+InjectFnmpRxPacket(_In_ FNMP_HANDLE Fnmp) {
+    const ETHERNET_ADDRESS localMac = FNMP_LOCAL_ETHERNET_ADDRESS_INIT;
+    const ETHERNET_ADDRESS remoteMac = FNMP_NEIGHBOR_ETHERNET_ADDRESS_INIT;
+
+    const UINT32 payloadBufferSize = 64'000;
+    UCHAR* payload = calloc(payloadBufferSize, sizeof(UCHAR));
```

Comment:
we need null checks for each allocation, or alternatively, we could preallocate enough buffers up front during queue creation.

Reply:
(this is a stress test, and things will start falling over, and we do not want our test to be one of them 😉 )